x-pack/plugins/triggers_actions_ui/public/application/sections/rules_list/components/rules_list.tsx

```diff
@@ -226,6 +232,7 @@ export const RulesList: React.FunctionComponent = () => {
     JSON.stringify(typesFilter),
     JSON.stringify(actionTypesFilter),
     JSON.stringify(ruleStatusesFilter),
+    JSON.stringify(tagsFilter),
```

Comment:
@JiaweiWu In case it is useful, here's what I used https://github.com/elastic/kibana/blob/main/x-pack/plugins/observability/public/hooks/use_fetch_rules.ts#L36 when I moved this code under observability in a `use_fetch_rules` hook. I got rid of JSON.stringify by using `useCallback`.

Reply:
Yep, this is getting improved in my rules_list refactor